diff --git a/src/cspice/err.c b/src/cspice/err.c
index e25d19f..a23409f 100644
--- a/src/cspice/err.c
+++ b/src/cspice/err.c
@@ -150,7 +150,7 @@ f__fatal(int n, char *s)
 	else
 		fprintf(stderr,"%s: %s\n",s,F_err[n-100]);
 	if (f__curunit) {
-		fprintf(stderr,"apparent state: unit %d ",f__curunit-f__units);
+		fprintf(stderr,"apparent state: unit %ld ",f__curunit-f__units);
 		fprintf(stderr, f__curunit->ufnm ? "named %s\n" : "(unnamed)\n",
 			f__curunit->ufnm);
 		}
-- 
1.5.6.3

