From 2d9d1d3075e4c1cd63a0f6d81039774a0c56e569 Mon Sep 17 00:00:00 2001
From: Mike Lundy <mike@fluffypenguin.org>
Date: Thu, 5 Feb 2009 14:29:12 -0800
Subject: [PATCH 3/6] append user CFLAGS

---
 configure.in |    2 +-
 1 files changed, 1 insertions(+), 1 deletions(-)

diff --git a/configure.in b/configure.in
index 529fe91..9eba9dc 100644
--- a/configure.in
+++ b/configure.in
@@ -126,7 +126,7 @@ pkgincludedir='$(includedir)/dc1394'
 AC_SUBST(pkgincludedir)
 
 dnl AX_CFLAGS_WARN_ALL(CFLAGS)
-CFLAGS="$CFLAGS -Wall -Wunused -std=gnu99"
+CFLAGS="-Wall -Wunused -std=gnu99 $CFLAGS"
 dnl for pedantic compilation, try to add "-pedantic" to the above string
 
 AC_CONFIG_FILES([ \
-- 
1.6.1

