From ec3420cb7f0b5e8cec761294f6b818e117ee2d47 Mon Sep 17 00:00:00 2001
From: Mike Lundy <mike@fluffypenguin.org>
Date: Thu, 5 Feb 2009 17:18:02 -0800
Subject: [PATCH 4/6] zero returned quad memory

---
 dc1394/linux/control.c |    3 +++
 1 files changed, 3 insertions(+), 0 deletions(-)

diff --git a/dc1394/linux/control.c b/dc1394/linux/control.c
index 787f981..04e4007 100644
--- a/dc1394/linux/control.c
+++ b/dc1394/linux/control.c
@@ -254,6 +254,9 @@ dc1394_linux_camera_read (platform_camera_t * cam, uint64_t offset,
 {
     int i, retval, retry = DC1394_MAX_RETRIES;
 
+    // zero memory to appease valgrind
+    memset(quads, 0, num_quads * sizeof(uint32_t));
+
     /* retry a few times if necessary (addition by PDJ) */
     while(retry--)  {
 #ifdef DC1394_DEBUG_LOWEST_LEVEL
-- 
1.6.1

