--- ./src/OpenThreads/CMakeLists.txt.org	2007-10-04 11:14:01.000000000 +0200
+++ ./src/OpenThreads/CMakeLists.txt	2008-01-05 16:59:57.000000000 +0100
@@ -5,9 +5,13 @@
 SET(OPENTHREADS_MAJOR_VERSION 2)
 SET(OPENTHREADS_MINOR_VERSION 2)
 SET(OPENTHREADS_PATCH_VERSION 0)
+IF(NOT DEFINED OPENTHREADS_SOVERSION)
 SET(OPENTHREADS_SOVERSION 9)
+ENDIF(NOT DEFINED OPENTHREADS_SOVERSION)
 
+IF(NOT DEFINED OPENTHREADS_VERSION)
 SET(OPENTHREADS_VERSION ${OPENTHREADS_MAJOR_VERSION}.${OPENTHREADS_MINOR_VERSION}.${OPENTHREADS_PATCH_VERSION})
+ENDIF(NOT DEFINED OPENTHREADS_VERSION)
 
 SET(HEADER_PATH ${OpenThreads_SOURCE_DIR}/include/OpenThreads)
 SET(OpenThreads_PUBLIC_HEADERS
--- ./src/OpenThreads/pthreads/CMakeLists.txt.org	2008-01-05 17:01:31.000000000 +0100
+++ ./src/OpenThreads/pthreads/CMakeLists.txt	2008-01-05 17:02:01.000000000 +0100
@@ -22,10 +22,6 @@
 	../common/Version.cpp
 )
 
-IF(OPENTHREADS_SONAMES)
-  SET_TARGET_PROPERTIES(${LIB_NAME} PROPERTIES VERSION ${OPENTHREADS_VERSION} SOVERSION ${OPENTHREADS_SOVERSION})
-ENDIF(OPENTHREADS_SONAMES)
-
 SET(CMAKE_REQUIRED_LIBRARIES_SAFE "${CMAKE_REQUIRED_LIBRARIES}")
 SET(CMAKE_REQUIRED_LIBRARIES "${CMAKE_REQUIRED_LIBRARIES} ${CMAKE_THREAD_LIBS_INIT}")
 
@@ -102,6 +98,10 @@
 	${CMAKE_CURRENT_BINARY_DIR}
 )
 
+IF(OPENTHREADS_SONAMES)
+  SET_TARGET_PROPERTIES(${LIB_NAME} PROPERTIES VERSION ${OPENTHREADS_VERSION} SOVERSION ${OPENTHREADS_SOVERSION})
+ENDIF(OPENTHREADS_SONAMES)
+
 INSTALL(
 	TARGETS OpenThreads
 	ARCHIVE DESTINATION lib${LIB_POSTFIX}
--- ./src/OpenThreads/pthreads/PThread.c++.org	2007-10-01 14:22:49.000000000 +0200
+++ ./src/OpenThreads/pthreads/PThread.c++	2008-01-05 17:04:25.000000000 +0100
@@ -590,27 +590,18 @@
 
     PThreadPrivateData *pd = static_cast<PThreadPrivateData *> (_prvData);
 
-    size_t defaultStackSize;
+    size_t defaultStackSize = 0;
     pthread_attr_getstacksize( &thread_attr, &defaultStackSize);
-    if(status != 0) {
-	return status;
-    }
 
     if(defaultStackSize < pd->stackSize) {
 
 	pthread_attr_setstacksize( &thread_attr, pd->stackSize);
-	if(status != 0) {
-	    return status;
-	}
     }
 
     //-------------------------------------------------------------------------
     // Now get what we actually have...
     //
     pthread_attr_getstacksize( &thread_attr, &defaultStackSize);
-    if(status != 0) {
-	return status;
-    }
 
     pd->stackSize = defaultStackSize;
 
--- ./src/OpenThreads/pthreads/PThreadPrivateData.h.org	2007-09-25 11:45:44.000000000 +0200
+++ ./src/OpenThreads/pthreads/PThreadPrivateData.h	2008-01-05 17:05:15.000000000 +0100
@@ -40,7 +40,7 @@
 
 private:
 
-    PThreadPrivateData() {};
+    PThreadPrivateData() : stackSize(0), stackSizeLocked(false), isRunning(false), isCanceled(false), idSet(false), tid(0), uniqueId(0), cpunum(0) {};
 
     virtual ~PThreadPrivateData() {};
 
