From 01ab151d0a1a3167455b051aee55aae152941aae Mon Sep 17 00:00:00 2001
From: Mike Lundy <mike@fluffypenguin.org>
Date: Sat, 31 Jan 2009 19:05:17 -0800
Subject: [PATCH 3/4] makefile

---
 GNUmakefile |    8 ++++----
 1 files changed, 4 insertions(+), 4 deletions(-)

diff --git a/GNUmakefile b/GNUmakefile
index 11421c1..5837493 100644
--- a/GNUmakefile
+++ b/GNUmakefile
@@ -228,12 +228,12 @@ ifeq ($(MACOSX_FRAMEWORK),yes)
 else
 	rm -f $(DESTDIR)$(INST_LIB)/$(GDAL_SLIB_B)
 	rm -f $(DESTDIR)$(INST_LIB)/$(GDAL_SLIB_B).$(GDAL_VERSION_MAJOR)
-	rm -f $(DESTDIR)$(INST_LIB)/$(GDAL_SLIB_B).$(GDAL_VER)
-	$(INSTALL_LIB) $(GDAL_SLIB) $(DESTDIR)$(INST_LIB)/$(GDAL_SLIB_B).$(GDAL_VER)
+	rm -f $(DESTDIR)$(INST_LIB)/$(GDAL_SLIB_B).$(GDAL_SLIB_SO_VER)
+	$(INSTALL_LIB) $(GDAL_SLIB) $(DESTDIR)$(INST_LIB)/$(GDAL_SLIB_B).$(GDAL_SLIB_SO_VER)
 	(cd $(DESTDIR)$(INST_LIB) ; \
-	 ln -s $(GDAL_SLIB_B).$(GDAL_VER) $(GDAL_SLIB_B).$(GDAL_VERSION_MAJOR))
+	 ln -s $(GDAL_SLIB_B).$(GDAL_SLIB_SO_VER) $(GDAL_SLIB_B))
 	(cd $(DESTDIR)$(INST_LIB) ; \
-	 ln -s $(GDAL_SLIB_B).$(GDAL_VERSION_MAJOR) $(GDAL_SLIB_B))
+	 ln -s $(GDAL_SLIB_B).$(GDAL_SLIB_SO_VER) $(GDAL_SLIB_B).$(GDAL_VERSION_MAJOR))
 endif
 
 else
-- 
1.6.1

