From 192f76d3a080471cedc479a9538ee61b45db58d8 Mon Sep 17 00:00:00 2001
From: Mike Lundy <mike@fluffypenguin.org>
Date: Thu, 2 May 2013 14:23:18 -0700
Subject: [PATCH] make grep not stupid.

---
 bin/zkServer.sh | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/bin/zkServer.sh b/bin/zkServer.sh
index aa6cbaa..e3edd44 100755
--- a/bin/zkServer.sh
+++ b/bin/zkServer.sh
@@ -81,7 +81,7 @@ fi
 echo "Using config: $ZOOCFG" >&2
 
 if [ -z $ZOOPIDFILE ]; then
-    ZOO_DATADIR=$(grep "^[[:space:]]*dataDir" "$ZOOCFG" | sed -e 's/.*=//')
+    ZOO_DATADIR=$(grep "^[[:space:]]*dataDir\>" "$ZOOCFG" | sed -e 's/.*=//')
     if [ ! -d "$ZOO_DATADIR" ]; then
         mkdir -p "$ZOO_DATADIR"
     fi
@@ -163,7 +163,7 @@ status)
     # -q is necessary on some versions of linux where nc returns too quickly, and no stat result is output
     STAT=`$JAVA "-Dzookeeper.log.dir=${ZOO_LOG_DIR}" "-Dzookeeper.root.logger=${ZOO_LOG4J_PROP}" \
              -cp "$CLASSPATH" $JVMFLAGS org.apache.zookeeper.client.FourLetterWordMain localhost \
-             $(grep "^[[:space:]]*clientPort" "$ZOOCFG" | sed -e 's/.*=//') srvr 2> /dev/null    \
+             $(grep "^[[:space:]]*clientPort\>" "$ZOOCFG" | sed -e 's/.*=//') srvr 2> /dev/null    \
           | grep Mode`
     if [ "x$STAT" = "x" ]
     then
-- 
1.8.2.1

