From 00bbdb931ab328a0f0f40e77bd6f438102768dc3 Mon Sep 17 00:00:00 2001
From: Mike Lundy <mike@fluffypenguin.org>
Date: Sat, 9 Jan 2010 04:33:28 -0800
Subject: [PATCH] fix jpeg2pdf strict aliasing bug

---
 tiff-4.0.0beta5/tools/tiff2pdf.c |   15 +++++++++------
 1 files changed, 9 insertions(+), 6 deletions(-)

diff --git a/tiff-4.0.0beta5/tools/tiff2pdf.c b/tiff-4.0.0beta5/tools/tiff2pdf.c
index d5f5848..1603fb5 100644
--- a/tiff-4.0.0beta5/tools/tiff2pdf.c
+++ b/tiff-4.0.0beta5/tools/tiff2pdf.c
@@ -5147,13 +5147,16 @@ tsize_t t2p_write_pdf_trailer(T2P* t2p, TIFF* output)
 	tsize_t written = 0;
 	char buffer[32];
 	int buflen = 0;
-	char fileidbuf[16];
+	union {
+		char bytes[16];
+		int  ints[4];
+	} fileidbuf;
 	int i = 0;
 
-	((int*)fileidbuf)[0] = rand();
-	((int*)fileidbuf)[1] = rand();
-	((int*)fileidbuf)[2] = rand();
-	((int*)fileidbuf)[3] = rand();
+	fileidbuf.ints[0] = rand();
+	fileidbuf.ints[1] = rand();
+	fileidbuf.ints[2] = rand();
+	fileidbuf.ints[3] = rand();
 	t2p->pdf_fileid = (unsigned char*)_TIFFmalloc(33);
 	if(t2p->pdf_fileid == NULL) {
 		TIFFError(
@@ -5166,7 +5169,7 @@ tsize_t t2p_write_pdf_trailer(T2P* t2p, TIFF* output)
 	_TIFFmemset(t2p->pdf_fileid, 0x00, 33);
 	for (i = 0; i < 16; i++) {
 		sprintf((char *)t2p->pdf_fileid + 2 * i,
-			"%.2hhX", fileidbuf[i]);
+			"%.2hhX", fileidbuf.bytes[i]);
 	}
 	written += t2pWriteFile(output, (tdata_t) "trailer\n<<\n/Size ", 17);
 	buflen = sprintf(buffer, "%lu", (unsigned long)(t2p->pdf_xrefcount+1));
-- 
1.6.6

