From 82a2470bd2ea5875ebe1f398abcac66af11a409d Mon Sep 17 00:00:00 2001
From: Mike Lundy <mike@fluffypenguin.org>
Date: Fri, 16 Sep 2011 19:01:16 -0700
Subject: [PATCH] remove-setjmp

---
 libunwind-1.0.1/src/Makefile.am |    2 +-
 1 files changed, 1 insertions(+), 1 deletions(-)

diff --git a/libunwind-1.0.1/src/Makefile.am b/libunwind-1.0.1/src/Makefile.am
index 26b1895..54ad5b4 100644
--- a/libunwind-1.0.1/src/Makefile.am
+++ b/libunwind-1.0.1/src/Makefile.am
@@ -480,7 +480,7 @@ endif # ARCH_ARM
 # libunwind-setjmp depends on libunwind-$(arch). Therefore must be added
 # at the end.
 if !REMOTE_ONLY
-lib_LTLIBRARIES += libunwind-setjmp.la
+#lib_LTLIBRARIES += libunwind-setjmp.la
 endif
 
 #
-- 
1.7.6.1

