From caf207f8fc496e0a3d2cb6071501ae3018c95d6c Mon Sep 17 00:00:00 2001
From: Mike Lundy <mike@fluffypenguin.org>
Date: Mon, 23 Feb 2009 01:44:10 -0700
Subject: [PATCH] destdir

---
 Makefile.am |   10 +++++-----
 1 files changed, 5 insertions(+), 5 deletions(-)

diff --git a/Makefile.am b/Makefile.am
index f717898..34b3b65 100644
--- a/Makefile.am
+++ b/Makefile.am
@@ -1,6 +1,6 @@
 AUTOMAKE_OPTIONS = foreign no-dependencies
 
-AM_CFLAGS = -DPACKAGE_DATA_DIR=\"$(pkgdatadir)\" @FFMPEG_CFLAGS@ @GLADE_CFLAGS@
+AM_CFLAGS = -DPACKAGE_DATA_DIR=\"$(DESTDIR)/$(pkgdatadir)\" @FFMPEG_CFLAGS@ @GLADE_CFLAGS@
 
 bin_PROGRAMS = electricsheep electricsheep-preferences
 
@@ -22,10 +22,10 @@ GNOME_SCREENSAVER_DATADIR = $(shell pkg-config --variable themesdir gnome-screen
 GNOME_SCREENSAVER_PRIVEXEDIR = $(shell pkg-config --variable privlibexecdir gnome-screensaver)
 
 install-data-local:
-	-$(INSTALL) electricsheep.desktop $(GNOME_SCREENSAVER_DATADIR)
-	-$(INSTALL) electricsheep-saver $(GNOME_SCREENSAVER_PRIVEXEDIR)
-	-$(INSTALL) -d $(pkgdatadir)
-	-$(INSTALL) 16_electricsheep /usr/share/gconf/defaults
+	-$(INSTALL) electricsheep.desktop $(DESTDIR)/$(GNOME_SCREENSAVER_DATADIR)
+	-$(INSTALL) electricsheep-saver $(DESTDIR)/$(GNOME_SCREENSAVER_PRIVEXEDIR)
+	-$(INSTALL) -d $(DESTDIR)/$(pkgdatadir)
+	-$(INSTALL) 16_electricsheep $(DESTDIR)/usr/share/gconf/defaults
 	-update-gconf-defaults
 
 uninstall-local:
-- 
1.6.1.2

