From f7f318ccbea2f7400dcfb3cad4c64191ae8e4d28 Mon Sep 17 00:00:00 2001
From: Mike Lundy <mike@fluffypenguin.org>
Date: Mon, 23 Feb 2009 01:21:14 -0700
Subject: [PATCH 2/3] remove crap

---
 configure.in |    4 ----
 1 files changed, 0 insertions(+), 4 deletions(-)

diff --git a/configure.in b/configure.in
index 5f5d810..d9c1029 100644
--- a/configure.in
+++ b/configure.in
@@ -19,9 +19,6 @@ AC_CHECK_LIB([expat],[XML_ParserCreate],,[AC_MSG_ERROR([The expat xml library is
 AC_FUNC_SETPGRP
 AC_CHECK_FUNCS(setproctitle)
 
-X11_PREFIX=`pkg-config --print-errors --variable=prefix x11`
-X11_SHARE="$X11_PREFIX/share"
-
 PKG_CHECK_MODULES([GLADE],  libglade-2.0 >= 2.5.0 gtk+-2.0 >= 2.7.0)
 PKG_CHECK_MODULES([FFMPEG], libavformat libavcodec libavutil)
 AC_SUBST(GLADE_CFLAGS)
@@ -29,7 +26,6 @@ AC_SUBST(GLADE_LIBS)
 AC_SUBST(FFMPEG_CFLAGS)
 AC_SUBST(FFMPEG_LIBS)
 
-AC_SUBST(X11_SHARE)
 AC_CONFIG_FILES([Makefile ])
 AC_CONFIG_FILES([electricsheep.desktop ])
 AC_CONFIG_FILES([electricsheep.xml ])
-- 
1.6.1.2

