From 0677215f7db2894e7e64499ddd7b4640c3fec157 Mon Sep 17 00:00:00 2001
From: Mike Lundy <mike@fluffypenguin.org>
Date: Mon, 23 Feb 2009 01:08:24 -0700
Subject: [PATCH 3/3] size_t is a long

---
 electricsheep.c |    2 +-
 1 files changed, 1 insertions(+), 1 deletions(-)

diff --git a/electricsheep.c b/electricsheep.c
index 175cc32..e0ee263 100755
--- a/electricsheep.c
+++ b/electricsheep.c
@@ -1775,7 +1775,7 @@ void update_server_sheep() {
 		    (unsigned)XML_GetCurrentLineNumber(parser));
 	    break;
 	}
-	if (debug > 1) fprintf(logout, "list read loop len=%d\n", len);
+	if (debug > 1) fprintf(logout, "list read loop len=%ld\n", len);
     } while (!done);
     XML_ParserFree(parser);
 
-- 
1.6.1.2

