diff -ur xmonad-contrib-darcs-1-orig/XMonad/Prompt/Window.hs xmonad-contrib-darcs-1/XMonad/Prompt/Window.hs
--- xmonad-contrib-darcs-1-orig/XMonad/Prompt/Window.hs	2009-01-24 23:54:26.000000000 -0800
+++ xmonad-contrib-darcs-1/XMonad/Prompt/Window.hs	2009-01-24 23:56:46.000000000 -0800
@@ -88,8 +88,11 @@
       gotoAction       = winAction W.focusWindow
       bringAction      = winAction bringWindow
       bringCopyAction  = winAction bringCopyWindow
+      filterP
+        | (subString c) = isInfixOf
+        | otherwise     = isPrefixOf
 
-      compList m s = return . filter (isPrefixOf s) . map fst . M.toList $ m
+      compList m s = return . filter (filterP s) . map fst . M.toList $ m
 
 
 -- | Brings a copy of the specified window into the current workspace.
diff -ur xmonad-contrib-darcs-1-orig/XMonad/Prompt.hs xmonad-contrib-darcs-1/XMonad/Prompt.hs
--- xmonad-contrib-darcs-1-orig/XMonad/Prompt.hs	2009-01-24 23:54:26.000000000 -0800
+++ xmonad-contrib-darcs-1/XMonad/Prompt.hs	2009-01-24 23:59:46.000000000 -0800
@@ -122,6 +122,7 @@
         , autoComplete      :: Maybe Int  -- ^ Just x: if only one completion remains, auto-select it,
         , showCompletionOnTab :: Bool     -- ^ Only show list of completions when Tab was pressed
                                           --   and delay by x microseconds
+        , subString         :: Bool       -- Do an infix search
         }
 
 data XPType = forall p . XPrompt p => XPT p
@@ -192,7 +193,8 @@
         , historyFilter     = id
         , defaultText       = []
         , autoComplete      = Nothing
-        , showCompletionOnTab = False }
+        , showCompletionOnTab = False
+        , subString         = False }
 greenXPConfig = defaultXPConfig { fgColor = "green", bgColor = "black" }
 amberXPConfig = defaultXPConfig { fgColor = "#ca8f2d", bgColor = "black", fgHLight = "#eaaf4c" }
 
