From f4740a5326f2bf586cf4e58228c3e2ab85593800 Mon Sep 17 00:00:00 2001
From: Mike Lundy <mike@fluffypenguin.org>
Date: Wed, 13 May 2009 14:29:45 -0400
Subject: [PATCH] fix double-free

---
 gaia-0.1.2/programs/gaia/TextureTile.cc |    1 -
 1 files changed, 0 insertions(+), 1 deletions(-)

diff --git a/gaia-0.1.2/programs/gaia/TextureTile.cc b/gaia-0.1.2/programs/gaia/TextureTile.cc
index a19536b..0e5308e 100644
--- a/gaia-0.1.2/programs/gaia/TextureTile.cc
+++ b/gaia-0.1.2/programs/gaia/TextureTile.cc
@@ -71,7 +71,6 @@ void TextureTile::Load(RawBuffer *data, int keep) {
 		tex->LoadJPEG(TEXTURE_SOURCE_MEM, data->Data(), data->Size());
 	} catch (...) {
 		delete tex;
-		delete data;
 		throw;
 	}
 
-- 
1.6.2

