From 0386f39fbb8f6de1bc8d8b441f9da2d9e4fdc489 Mon Sep 17 00:00:00 2001
From: Mike Lundy <mike@fluffypenguin.org>
Date: Tue, 29 Sep 2009 11:47:57 -0700
Subject: [PATCH] fix empty conditionals with autoconf 2.64

---
 ACE_wrappers/configure.ac |    8 ++++----
 1 files changed, 4 insertions(+), 4 deletions(-)

diff --git a/ACE_wrappers/configure.ac b/ACE_wrappers/configure.ac
index 4e116f8..994c55f 100644
--- a/ACE_wrappers/configure.ac
+++ b/ACE_wrappers/configure.ac
@@ -2679,7 +2679,7 @@ dnl END OUTER REQUIRE PRAGMA #########################################
       AC_DEFINE([ACE_TEMPLATES_REQUIRE_PRAGMA])
      ],
      [
-      dnl Do nothing.
+      : dnl Do nothing.
      ])
   ])
 
@@ -4617,7 +4617,7 @@ if test "$ace_user_enable_reentrant_funcs" = yes; then
                   fi
                  ],
                  [
-                  dnl Nothing to do!
+                  : dnl Nothing to do!
                  ])
                 ],
                 [AC_DEFINE([ACE_LACKS_PWD_REENTRANT_FUNCTIONS])])
@@ -5660,7 +5660,7 @@ ACE_CACHE_CHECK([if msgsnd() takes a struct msgbuf* second argument],
          ])
      ],
      [
-      dnl Do nothing if msgsnd takes a const void* second argument
+      : dnl Do nothing if msgsnd takes a const void* second argument
      ],
      [
       dnl If we get this far we presumably have a non-const void* second param
@@ -7205,7 +7205,7 @@ ACE_CACHE_CHECK([if generated ACE configuration is usable],
      ])
   ],
   [
-   dnl Looks good! Do nothing.
+   : dnl Looks good! Do nothing.
    dnl It appears that ace/OS.cpp compiled.  If it didn't compile then
    dnl there would be no chance that the rest of ACE would compile.
   ],
-- 
1.6.4

