diff -ur ACE_wrappers-orig/TAO/TAO_IDL/be/be_helper.cpp ACE_wrappers/TAO/TAO_IDL/be/be_helper.cpp
--- ACE_wrappers-orig/TAO/TAO_IDL/be/be_helper.cpp	2008-04-30 05:15:49.000000000 -0700
+++ ACE_wrappers/TAO/TAO_IDL/be/be_helper.cpp	2009-01-12 17:10:36.000000000 -0800
@@ -360,7 +360,7 @@
 
   return *this;
 }
-#if defined (ACE_WIN64) || ! (! defined (ACE_HAS_UINT64_T) && ACE_SIZEOF_LONG == 8)
+#if defined (ACE_WIN64) || (defined (ACE_HAS_UINT64_T) && ACE_SIZEOF_LONG != ACE_SIZEOF_LONG_LONG)
 TAO_OutStream &
 TAO_OutStream::operator<< (const ACE_CDR::ULongLong num)
 {
Only in ACE_wrappers/TAO/TAO_IDL/be_include: .be_helper.h.swp
diff -ur ACE_wrappers-orig/TAO/TAO_IDL/be_include/be_helper.h ACE_wrappers/TAO/TAO_IDL/be_include/be_helper.h
--- ACE_wrappers-orig/TAO/TAO_IDL/be_include/be_helper.h	2008-04-30 05:15:49.000000000 -0700
+++ ACE_wrappers/TAO/TAO_IDL/be_include/be_helper.h	2009-01-12 17:11:30.000000000 -0800
@@ -174,7 +174,7 @@
 
   TAO_OutStream &operator<< (const ACE_CDR::Long num);
   // output the integer and return a reference to ourselves
-#if defined (ACE_WIN64) || ! (! defined (ACE_HAS_UINT64_T) && ACE_SIZEOF_LONG == 8)
+#if defined (ACE_WIN64) || (defined (ACE_HAS_UINT64_T) && ACE_SIZEOF_LONG != ACE_SIZEOF_LONG_LONG)
   TAO_OutStream &operator<< (const ACE_CDR::ULongLong num);
   // output the integer and return a reference to ourselves
 
